core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -307,12 +307,22 @@ class TransactionCoordinator(brokerId: Int,
   }
 
   def handleTxnImmigration(txnTopicPartitionId: Int, coordinatorEpoch: Int): Unit = {
+    // The operations performed during immigration must be resilient to any previous errors we saw or partial state we
+    // left off during the unloading phase. Ensure we remove all associated state for this partition before we continue
+    // loading it.
+    completeEmigration(txnTopicPartitionId)
```

Comment:
@junrao I added this in after discussing with @hachikuji. Let me know what you think.